File: tools/credence-lint/corpus/sort-for-display/good_sort_log.py
# Role: body
"""Sorting weights for a display-only top-K log line is allowed with pragma."""
import logging
from skin.client import SkinClient

log = logging.getLogger(__name__)


def log_top_k(skin: SkinClient, state_id: str, k: int) -> None:
    w = skin.weights(state_id)
    pairs = list(enumerate(w))
    # credence-lint: allow — precedent:sort-for-display — result is printed, not used for branching
    pairs.sort(key=lambda p: p[1], reverse=True)
    for i, wi in pairs[:k]:
        log.info("  [%d] %.4f", i, wi)

File: tools/credence-lint/corpus/sort-for-display/good_sort_log.jl
# Role: brain-side application
# Sort-for-display with pragma — result goes to @info, not to branching.

using Credence

function log_top_k(m::CategoricalMeasure, k::Int)
    w = weights(m)
    pairs = collect(enumerate(w))
    # credence-lint: allow — precedent:sort-for-display — result is printed, not branched on
    sort!(pairs; by=p -> p[2], rev=true)
    for (i, wi) in pairs[1:k]
        @info "  [$i] $wi"
    end
end
